domain/src/main/java/org/oppia/domain/oppialogger/logstorage/LogReportStorageHelper.kt

```diff
@@ -0,0 +1,172 @@
+package org.oppia.domain.oppialogger.logstorage
+
+import androidx.lifecycle.LiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.ExceptionLog
+import org.oppia.app.model.OppiaEventLogs
+import org.oppia.app.model.OppiaExceptionLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.domain.oppialogger.EventLogStorageCacheSize
+import org.oppia.domain.oppialogger.ExceptionLogStorageCacheSize
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.ConsoleLogger
+import org.oppia.util.threading.BackgroundDispatcher
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Helper class for storing log reports in absence of network connectivity. */
+@Singleton
+class LogReportStorageHelper @Inject constructor(
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val dataProviders: DataProviders,
+  private val consoleLogger: ConsoleLogger,
+  @BackgroundDispatcher private val backgroundCoroutineDispatcher: CoroutineDispatcher,
+  @EventLogStorageCacheSize private val eventLogStorageCacheSize: Int,
+  @ExceptionLogStorageCacheSize private val exceptionLogStorageCacheSize: Int
+) {
+  private enum class LogReportType {
+    EVENT_LOG,
+    EXCEPTION_LOG
+  }
+  private val eventLogStore =
+    cacheStoreFactory.create("event_logs", OppiaEventLogs.getDefaultInstance())
+  private val exceptionLogStore =
+    cacheStoreFactory.create("exception_logs", OppiaExceptionLogs.getDefaultInstance())
```

Comment:
Based on my top-level PR comment, I do wonder whether we can just split this helper into two and merge the offline caching functionality into the corresponding controller that needs it. There doesn't seem to be any logical overlap here, and the current implementation seems to be doing a bit too much trying manage both the event and exception log (it feels a bit like a violation of the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)).

Reply:
Agreed and modified accordingly. PTAL :)